docs/source/utils/notebook.py

```diff
@@ -1,42 +1,157 @@
 \"\"\"
 
 # %%
-!python3 -m pip install -q dff[tutorials]
-# Installs dff with dependencies for running tutorials
-# To install the minimal version of dff, use `pip install dff`
-# To install other options of dff, use `pip install dff[OPTION_NAME1,OPTION_NAME2]`
-# where OPTION_NAME can be one of the options from EXTRA_DEPENDENCIES.
-# e.g `pip install dff[ydb, mysql]` installs dff with dependencies for using Yandex Database and MySQL
-# EXTRA_DEPENDENCIES can be found in
-# https://github.com/deeppavlov/dialog_flow_framework/blob/dev/README.md#installation
-
+%pip install {matchobj.group(1)}
 
 # %% [markdown]
 \"\"\"
 __Running tutorial__
 \"\"\"
+"""
+
+
+class DocumentationLink(ReplacePattern):
+    """
+    Replace documentation linking directives.
+
+    Replace strings of the `!doclink({args})` format with corresponding links to local files.
+
+    `args` is a comma-separated string of arguments to pass to the :py:meth:`.DocumentationLink.link_to_doc_page`.
+
+    So, `!doclink(arg1,arg2,arg3)` will be replaced with `link_to_doc_page(arg1, arg2, arg3)`, and
+    `!doclink(arg1,arg2)` will be replaced with `link_to_doc_page(arg1, arg2)`.
+
+    USAGE EXAMPLES
+    --------------
+
+    [link](!doclink(api,script.core.script))
+
+    [link](!doclink(api,script.core.script,Node))
+
+    [link](!doclink(tutorial,messengers.web_api_interface.4_streamlit_chat))
+
+    [link](!doclink(tutorial,messengers.web_api_interface.4_streamlit_chat,API-configuration))
+
+    [link](!doclink(guide,basic_conceptions))
+
+    [link](!doclink(guide,basic_conceptions,example-conversational-chat-bot))
+
+    """
+
+    pattern: ClassVar[re.Pattern] = re.compile(r"!doclink\((.+?)\)")
+
+    @staticmethod
+    def link_to_doc_page(
+        page_type: Literal["api", "tutorial", "guide"],
+        page: str,
+        anchor: Optional[str] = None,
+    ):
+        """
+        Create a link to a documentation page.
+
+        :param page_type:
+            Type of the documentation:
+
+                - "api" -- API reference
+                - "tutorial" -- Tutorials
+                - "guide" -- User guides
+
+        :param page:
+            Name of the page without the common prefix.
+
+            So, to link to keywords, pass "script.core.keywords" as page (omitting the "dff" prefix).
+
+            To link to the basic script tutorial, pass "script.core.1_basics" (without the "tutorials" prefix).
+
+            To link to the basic concepts guide, pass "basic_conceptions".
+        :param anchor:
+            An anchor on the page. (optional)
+
+            For the "api" type, use only the last part of the linked object.
+
+            So, to link to the `CLIMessengerInterface` class, pass "CLIMessengerInterface" only.
+
+            To link to a specific section of a tutorial or a guide, pass an anchor as-is (e.g. "introduction").
+        :return:
+            A link to the corresponding documentation part.
+        """
+        doc_prefix = ".."
```

Comment:
This might well be a file-level constant instead (or a default parameter). I'm not sure that it will always behave in the same way on imports.